### PR TITLE
comparison of integer expressions of different signedness issue

### LINKED ISSUE #3376: 'bool rz_buf_read_offset(RzBuffer*, long long unsigned int*, unsigned char*, size_t)': rz_util/rz_buf.h:238:54: warning: comparison of integer expressions of different signedness: 'long long int' and 'size_t'
When compiling Cutter we get hundred of the following warnings. Would be nice to address them:
```
                 from /tmp/cutter/src/core/Cutter.h:4,
/tmp/cutter/build/Rizin-prefix/include/librz/rz_util/rz_buf.h: In function 'bool rz_buf_read_offset(RzBuffer*, long long unsigned int*, unsigned char*, size_t)':
/tmp/cutter/build/Rizin-prefix/include/librz/rz_util/rz_buf.h:238:54: warning: comparison of integer expressions of different signedness: 'long long int' and 'size_t' {aka 'long unsigned int'} [-Wsign-compare]
  238 |         if (rz_buf_read_at(b, *offset, result, size) != size) {
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
/tmp/cutter/build/Rizin-prefix/include/librz/rz_util/rz_buf.h: In function 'bool rz_buf_write_offset(RzBuffer*, long long unsigned int*, unsigned char*, size_t)':
/tmp/cutter/build/Rizin-prefix/include/librz/rz_util/rz_buf.h:257:55: warning: comparison of integer expressions of different signedness: 'long long int' and 'size_t' {aka 'long unsigned int'} [-Wsign-compare]
  257 |         if (rz_buf_write_at(b, *offset, result, size) != size) {
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~

```

### PR DESCRIPTION
**Pull request by FaresAtef1**
- Fixes issue #3376 

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

I have added a check and parsing to resolve the warnings.

**Test plan**

...

**Closing issues**

closes #3376

### CODE REVIEW

File: librz/include/rz_util/rz_buf.h [hunk 1]
```diff
@@ -235,7 +235,11 @@ DEFINE_RZ_BUF_WRITE_BLE(64)
 
 static inline bool rz_buf_read_offset(RZ_NONNULL RzBuffer *b, RZ_NONNULL RZ_INOUT ut64 *offset, RZ_NONNULL RZ_OUT ut8 *result, size_t size) {
 	rz_return_val_if_fail(b && offset && result, false);
-	if (rz_buf_read_at(b, *offset, result, size) != size) {
+	long long temp=rz_buf_read_at(b, *offset, result, size);
+	if(temp<0)
+		return false;
+	size_t temp_t=(size_t) temp;
+	if (temp != size) {
```
Comment (wargio) [inline librz/include/rz_util/rz_buf.h]: Thanks for the contrib, in our project we always use brackets to define scopes (if/else/while/for/etc..)
You don't really need to make this changes, just cast inline.
Also on rizin we have declarations for `long long`, etc.. just use `st64` for signed.
I suggest to just change the line from
`if (rz_buf_read_at(b, *offset, result, size) != size) {`
to
`if (rz_buf_read_at(b, *offset, result, size) != (st64)size) {`

**Review by wargio — CHANGES_REQUESTED** [review]
see comment.

**Comment by wargio** [comment]
one last request. each time you push a PR, please before committing, run clang format, and this is how:

1. Install clang-format (some distros bundles it with clang)
2. From the root folder of the rizin project run the following command: `./sys/clang-format.py`
3. Commit and push
4. Repeat step 2 before committing any file.

File: librz/include/rz_util/rz_buf.h (hunk 1 above)
Comment (FaresAtef1) [reply librz/include/rz_util/rz_buf.h]: Done

**Review by wargio — APPROVED** [review]

**Pull request merged**